src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -680,19 +693,28 @@ public void ReportProgress(int percentProgress, string text, string description,
         protected virtual void OnDoWork(DoWorkEventArgs e)
         {
             DoWorkEventHandler handler = DoWork;
-            if (!(handler is null))
+            if (handler is null)
             {
-                var eventArgs = new ProgressDialogDoWorkEventArgs(e.Argument, CancellationToken.None)
+                return;
+            }
+
+            using (var cts = System.Threading.CancellationTokenSource.CreateLinkedTokenSource(_cancellationToken))
```

Comment:
This needs to move to the `Show` / `ShowDialog` - we'd create a new `CancellationTokenSource` via `CreateLinkedTokenSource` at that time, using the `CancellationToken` provided.